include/cuco/detail/trie/dynamic_bitset/dynamic_bitset.cuh

```diff
@@ -0,0 +1,359 @@
+   * @param keys_end End iterator to keys list
+   * @param outputs_begin Begin iterator to outputs selects list
+   * @param stream Stream to execute selects kernel
+   */
+  template <typename KeyIt, typename OutputIt>
+  constexpr void selects(KeyIt keys_begin,
+                         KeyIt keys_end,
+                         OutputIt outputs_begin,
+                         cuda_stream_ref stream = {}) const noexcept;
+
+  /**
+   *@brief Struct to hold all storage refs needed by bitvector_ref
+   */
+  // TODO: this is not a real ref type, to be changed
+  struct storage_ref_type {
+    const slot_type* words_ref_;  ///< Words refs
+
+    const rank* ranks_ref_;         ///< Ranks refs
+    const size_type* selects_ref_;  ///< Selects refs
+
+    const rank* ranks0_ref_;         ///< Ranks refs for 0 bits
+    const size_type* selects0_ref_;  ///< Selects refs 0 bits
+  };
+
+  /**
+   * @brief Device non-owning reference type of dynamic_bitset
+   */
+  class reference {
+   public:
+    /**
+￼   * @brief Constructs dynamic_bitset_ref.
+￼   *
+￼   * @param storage Struct with non-owning refs to bitvector slot storages
+￼   */
+    __host__ __device__ explicit constexpr reference(storage_ref_type storage) noexcept;
+
+    /**
+     * @brief Access value of a single bit
+     *
+     * @param key Position of bit
+     *
+     * @return Value of bit at position specified by key
+     */
+    [[nodiscard]] __device__ constexpr bool get(size_type key) const noexcept;
+
+    /**
+     * @brief Access a single word of internal storage
+     *
+     * @param word_id Index of word
+     *
+     * @return Word at position specified by index
+     */
+    [[nodiscard]] __device__ constexpr slot_type get_word(size_type word_id) const noexcept;
+
+    /**
+     * @brief Find position of first set bit starting from a given position (inclusive)
+     *
+     * @param key Position of starting bit
+     *
+     * @return Index of next set bit
+     */
+    [[nodiscard]] __device__ size_type find_next_set(size_type key) const noexcept;
+
+    /**
+     * @brief Find number of set bits (rank) in all positions before the input position (exclusive)
+     *
+     * @param key Input bit position
+     *
+     * @return Rank of input position
+     */
+    [[nodiscard]] __device__ constexpr size_type rank(size_type key) const noexcept;
+
+    /**
+     * @brief Find position of Nth set (1) bit counting from start of bitvector
+     *
+     * @param count Input N
+     *
+     * @return Position of Nth set bit
+     */
+    [[nodiscard]] __device__ constexpr size_type select(size_type count) const noexcept;
+
+    /**
+     * @brief Find position of Nth not-set (0) bit counting from start of bitvector
+     *
+     * @param count Input N
+     *
+     * @return Position of Nth not-set bit
+     */
+    [[nodiscard]] __device__ constexpr size_type select0(size_type count) const noexcept;
```

Comment:
```suggestion
    [[nodiscard]] __device__ constexpr size_type select_false(size_type count) const noexcept;
```
`select_false` instead of `select0` based on the offline discussion

Reply:
c7faed0992d792f37919b8749949d0d3e7cf88ee